File: Ejercicios en clase/Arrreglos/Array2d.java
package Arrreglos;

public class Array2d {
    Character[][] data;
    int rowSize;
    int colSize;

    public Array2d(int ren, int col) {
        this.rowSize = ren;
        this.colSize = col;
        this.data = new Character[ren][col];
    }

    public int getRowSize() {
        return rowSize;
    }

    public void setRowSize(int rowSize) {
        this.rowSize = rowSize;
    }

    public int getColSize() {
        return colSize;
    }

    public void setColSize(int colSize) {
        this.colSize = colSize;
    }

    public void setItem(int ren, int col, Character dato) {
        if (ren >= 0 && ren < this.rowSize && col >= 0 && col < this.colSize) {
            this.data[ren][col] = dato;
        } else {
            System.out.println("Indices fuera de rango");
        }

    }

    public Character getItem(int ren, int col) {
        if (ren >= 0 && ren < this.rowSize && col >= 0 && col < this.colSize) {
            return this.data[ren][col];
        } else {
            return '\0';
        }

    }

    public void clear(Character dato) {
        for (int i = 0; i < this.rowSize; i++) {
            for (int j = 0; j < data.length; j++) {
                this.data[i][j] = (dato);
            }
        }
    }

    @Override
    public String toString() {
        String str = "";

        for (int i = 0; i < this.rowSize; i++) {
            for (int j = 0; j < this.colSize; j++) {
                str = str + this.data[i][j] + ", ";
            }
            str = str + "\n";
        }
        return str;
    }

}

File: Laberinto/Array2D.py
class Array2d:
    def __init__(self, ren, col):
        self.rowSize = ren
        self.colSize = col
        self.data = [['' for _ in range(col)] for _ in range(ren)]

    def get_rowSize(self):
        return self.rowSize

    def get_colSize(self):
        return self.colSize

    def get_item(self, ren, col):
        if 0 <= ren < self.rowSize and 0 <= col < self.colSize:
            return self.data[ren][col]
        else:
            print("Índices fuera de rango.")

    def set_item(self, ren, col, dato):
        if 0 <= ren < self.rowSize and 0 <= col < self.colSize:
            self.data[ren][col] = dato
        else:
            print("Índices fuera de rango.")

    def clear(self, dato):
        for i in range(self.rowSize):
            for j in range(self.colSize):
                self.data[i][j] = dato

    def __str__(self):
        str_rep = ""
        for i in range(self.rowSize):
            for j in range(self.colSize):
                str_rep += f"{self.data[i][j]}, "
            str_rep += "\n"
        return str_rep
